docs/package_templates/header_only/all/conanfile.py

```diff
@@ -0,0 +1,114 @@
+    settings = "os", "arch", "compiler", "build_type" # even for header only
+    no_copy_source = True # do not copy sources to build folder for header only projects, unless, need to apply patches
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 14
+
+    # in case the project requires C++14/17/20/... the minimum compiler version should be listed
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "15",
+            "msvc": "14.1",
+            "gcc": "5",
+            "clang": "5",
+            "apple-clang": "5.1",
+        }
+
+    # no exports_sources attribute, but export_sources(self) method instead
+    # this allows finer grain exportation of patches per version
+    def export_sources(self):
+        for p in self.conan_data.get("patches", {}).get(self.version, []):
+            copy(self, p["patch_file"], self.recipe_folder, self.export_sources_folder)
+
+    def layout(self):
+        # src_folder must use the same source folder name the project
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        # prefer self.requires method instead of requires attribute
+        self.requires("dependency/0.8.1")
+
+    # same package ID for any package
+    def package_id(self):
+        self.info.clear()
+
+    def validate(self):
+        # compiler subsettings are not available when building with self.info.clear()
+        if self.info.settings.get_safe("compiler.cppstd"):
+            # validate the minimum cpp standard supported when installing the package. For C++ projects only
+            check_min_cppstd(self, self._minimum_cpp_standard)
+        minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
+        if minimum_version and Version(self.info.settings.get_safe("compiler.version")) < minimum_version:
+            raise ConanInvalidConfiguration(f"{self.ref} requires C++{self._minimum_cpp_standard}, which your compiler does not support.")
+        # in case it does not work in another configuration, it should validated here too
+        if self.info.settings.os == "Windows":
+            raise ConanInvalidConfiguration(f"{self.ref} can not be used on Windows.")
+
+    # if another tool than the compiler or CMake is required to build the project (pkgconf, bison, flex etc)
+    def build_requirements(self):
+        self.tool_requires("tool/x.y.z")
+
+    def source(self):
+        # download source package and extract to source folder
+        get(**self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    # not mandatory when there is no patch, but will suppress warning message about missing build() method
+    def build(self):
+        # apply patches in source_folder. The attribute no_copy_source should not be used when applying patches
```

Comment:
```suggestion
```

Reply:
My suggestion is keeping 'The attribute no_copy_source should not be used when applying patches'. So we can copy to the build folder, apply the patches and keep source folder immaculate.

Reply:
Yes, very important to remove `no_copy_source = True` in this case, otherwise 2 successive builds through conan create/install will try to apply patches in source folder, so the first one will modify files under source folder, and the second build will fail during this new attempt to patch files.

Reply:
I didn't not notice the second half 

```suggestion
        # The attribute no_copy_source should not be used when applying patches in build
```

Reply:
Done, applied last suggestion.